File: Service.py
import time
import sys
from agent import Event
from agent import Agent

class Service:
	def perform_service(self, agent_handler):
		pass
		
class Service1(Service):
	def perform_service(self, agent_handler):
		#performs some service
		print("Performing service 1")
		time.sleep(5)
		a = Agent()
		print(a)
		a.notify(Event.OP1)
		
class Service2(Service):
	def perform_service(self, agent_handler):
		#performs some service 
		time.sleep(10)
		agent_handler.notify(Event.OP2)

File: Service.java
import java.util.concurrent.TimeUnit;

abstract class Service
{
	private EventType e = EventType.OP;
	abstract public void performService();

	public EventType getEventType()
	{
		return e;
	}
}

class Service1 extends Service
{
	private static final EventType e = EventType.OP1;
	public void performService() 
	{
		try
		{
			TimeUnit.SECONDS.sleep(3);
		}
		catch (InterruptedException e)
		{
			System.out.println(e);
		}
		//singleton 
		Agent a = Agent.getInstance();
		a.notify(e);
	}

	public EventType getEventType()
	{
		return e;
	}
}

class Service2 extends Service
{
	private static final EventType e = EventType.OP2;
	public void performService()
	{
		Agent a = Agent.getInstance();
		a.notify(e);
	}

	public EventType getEventType()
	{
		return e;
	}
}